server/plugin.go

```diff
@@ -426,6 +428,36 @@ func (p *Plugin) OnActivate() error {
 		}
 	}
 
+	remoteID, err := p.API.RegisterPluginForSharedChannels(model.RegisterPluginOpts{
+		Displayname:  "ms_teams_connect",
+		PluginID:     pluginID,
+		CreatorID:    botID,
+		AutoShareDMs: true,
+		AutoInvited:  true,
+	})
+	if err != nil {
+		return err
+	}
+	p.remoteID = remoteID
+
+	linkedChannels, err := p.store.ListChannelLinks()
+	if err != nil {
+		return err
+	}
+	for _, linkedChannel := range linkedChannels {
+		p.API.LogDebug("LINKED CHANNEL", "channelID", linkedChannel.MattermostChannelID, "teamID", linkedChannel.MattermostTeamID, "remoteID", p.remoteID)
+		// TODO: Handling error here or using a better policy about re-share channels
```

Comment:
Sharing an already shared channel was made idempotent with no error.  See https://github.com/mattermost/mattermost/blob/eee5a71e3d18d04b1b136e44cd8d848fd6e3bea9/server/channels/app/plugin_api.go#L1309

Reply:
... meaning, if there is an error here, it is a real error such as invalid channel_id or database error.

Reply:
Then I going to add a "warning" log in case of error there.

Reply:
Done